core/src/main/java/edu/wpi/grip/core/sources/CameraSource.java

```diff
@@ -154,8 +154,21 @@ public void start() throws IOException, IllegalStateException {
 
                     frameMat.copyTo(frameOutputSocket.getValue().get());
                     frameOutputSocket.setValue(frameOutputSocket.getValue().get());
-                    long thisMoment = System.currentTimeMillis();
-                    frameRateOutputSocket.setValue(1000 / (thisMoment - lastFrame));
+
+                    final long thisMoment = System.nanoTime();
+                    final long elapsedTime = thisMoment - lastFrame;
+                    if (elapsedTime == 0) {
+                        /* This can happen when using system.currentTimeMillis():
+                         * https://github.com/WPIRoboticsProjects/GRIP/issues/199
+                         * As such we use system.nanoTime().
+                         * However, we add an assertion to ensure this bug doesn't reappear.
+                         */
+                        assert elapsedTime != 0 : "The time between grabbing two frames was less than 1 nanosecond";
+                        frameRateOutputSocket.setValue(frameRateOutputSocket.getValue().get());
+                    } else {
+                        assert elapsedTime > 0 : "Time has run backwards!";
```

Comment:
Right, but assertions are good because they'll tell us if we break something before we deploy.  `elapsedTime != 0` will tell us if an unlikely timing condition occurs, and since that would not happen consistently, the assertion would probably not catch it before deploying.

You're right that `assert elapsedTime > 0` should never happen, but I don't think that's even worth asserting.